src/_nebari/stages/nebari_tf_extensions/__init__.py

```diff
@@ -2,6 +2,9 @@
 from typing import Any, Dict, List
 
 from _nebari.stages.base import NebariTerraformStage
+from _nebari.stages.infrastructure import InfrastructureOutputSchema
```

Comment:
Is there a way we could make this system work without requiring imports? I'm still not sure if this is good/bad. I think having a discussion about this would give me a better idea on how this might work. I'm worried that plugins will have to import their dependencies in order to list one another as a dependency.

Reply:
I'm definitely open to discussing this in more detail. This was my attempt at making the dependencies more explicit. I understand that having explicit dependencies might place more restrictions on how stages are developed. 

That said, I would argue that we have implicit dependencies currently, namely `stage_outputs` and env vars set by the `contextmanager` of the previous sage. In stages developed outside of the main Nebari codebase, they might need to know which the exact values in `stage_outputs` (ie `stage_output['<stage_name>']['value']`).